### ISSUE TITLE
Search with English verbs is not producing results

### ISSUE THREAD

**Issue by aarppe** [UAlbertaALTLab/cree-intelligent-dictionary#158]
Searching with any English verb, such as eat, sleep, or see, isn't resulting in Cree verbs (though there are some occasional matches with other Cree lexical entries with the search word in their English glosses).

Example expectations for verbs (at the very least):

- [x] _see_ -> _wâpiw_, _wâpahtam_, _wâpamêw_
- [x] _eat_ -> _mîcisow_, _mîciw_, _mowêw_
- [x] _sleep_ -> _nipâw_

This doesn't seem to be a matter of prefix search not yet implemented for English.

**Comment by Madoshakalaka** [comment]
The search algorithm for English keywords needs to be redesigned. And I need some guidance. @eddieantonio @aarppe 

Currently, for an English keyword, the algorithm finds the English word in `engcrk.xml` and returns the lemmas in database that overlap **exact definition** with the English translation in `engcrk.xml`

Concretely, this long gigantic thing attached at the bottom is what's inside `engcrk.xml` for English word "see", see the bottom to get an idea of the data structure

So the algorithm uses all the "truncs" such as `<trunc>oh! That's it, its it? see.</trunc>` and returns lemmas that has exact definitions.

I have realised quite early many of those "truncs" are quite arbitrary, in that they don't match with database entries, which essentially comes from `crkeng.xml` (are they manually created by human, good effort @atticusha ). 

- Why don't you get "wapamew"? 
- simple. The database definition for "wapamew" has **s/he sees s.o., s/he witnesses s.o.** while the `trunc` for wapamew here says **see s.o., witness s.o.**, which is NOT an exact match.


```
<e>
   <lg xml:lang="eng">
      <l pos="V">see</l>
   </lg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">oh! That's it, its it? see.</trunc>
      <t rank="1.0" pos="Ipc">kah!</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">just wait and see.</trunc>
      <t rank="1.0" pos="Ipc">macikocican</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">come, let's see</trunc>
      <t rank="1.0" pos="Ipc">mahci</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">let me pass, please.</trunc>
      <trunc sources="CW">well, then, please; come, let's see [hortatory]</trunc>
      <t rank="1.0" pos="Ipc">mahti</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">look, you will see for yourself that it is so</trunc>
      <t rank="1.0" pos="Ipc">mâcikôci</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">look, let me show you; you will see</trunc>
      <t rank="1.0" pos="Ipc">mâcikôtitân</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">you will see for yourself that it is so</trunc>
      <t rank="1.0" pos="Ipc">mâcikôtân</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">good-bye, see you again; literally: &quot;I will see you again&quot;</trunc>
      <t rank="1.0" pos="Ipc">kîhtwâm ka-wâpamitin</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">please; let's see</trunc>
      <t rank="1.0" pos="Ipc">mahti êsa</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see here!</trunc>
      <t rank="1.0" pos="Ipc">cîst</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">back in time; far back; from before, earlier; having lived long enough for something; chance to, live to, live to see</trunc>
      <t rank="1.0" pos="Ipc">ocihci-</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">Opawikoscikan, SK; Woods Cree reserve; literally: &quot;at the little narrows&quot;; see Pelican Narrows</trunc>
      <t rank="1.0" pos="N">wapâsihk</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">1. little chief, boss; office clerk; prince; [in jest:] &quot;noble&quot; 2. Littlechief, personal and family name from White Bear First Nation, SK 3. Okemasis, personal name of Cree chief and Saskatchewan Reserve; see Beardy's and Okemasis</trunc>
      <t rank="1.0" pos="N">okimâsis</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">being alive to see one another</trunc>
      <t rank="1.0" pos="N">ocâpahtowin</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">can be easily seen. stand out in a crowd.</trunc>
      <t rank="1.0" pos="V">ekihkahnakosit</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">be hidden where you can not see it.</trunc>
      <t rank="1.0" pos="V">kacinakwan</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">be decorated or marked so as to be seen.</trunc>
      <t rank="1.0" pos="V">kiskinawacihaw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">le me see, please.</trunc>
      <t rank="1.0" pos="V">mahtesa</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">people keep going to see him.</trunc>
      <t rank="1.0" pos="V">nanitawapimaw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">keep going to see him.</trunc>
      <t rank="1.0" pos="V">nanitawapimew</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">live long enough to see it [again].</trunc>
      <t rank="1.0" pos="V">otisahpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">see it clearly.</trunc>
      <t rank="1.0" pos="V">pakiskinam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">not notice it. miss seeing it.</trunc>
      <t rank="1.0" pos="V">patahpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">come and see it.</trunc>
      <t rank="1.0" pos="V">pewapahta</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">come and see him.</trunc>
      <t rank="1.0" pos="V">pewapam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">be seen through. An x-ray is taken of him.</trunc>
      <t rank="1.0" pos="V">sapwapimaw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">see through him. take an x-ray of him.</trunc>
      <t rank="1.0" pos="V">sapwapimew</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">see wrongly.</trunc>
      <t rank="1.0" pos="V">wanapimow</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">see it.</trunc>
      <t rank="1.0" pos="V">wapahta</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">stay till morning. see the dawn or morning.</trunc>
      <t rank="1.0" pos="V">wapanisow</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">see him.</trunc>
      <t rank="1.0" pos="V">wapim</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">be seen.</trunc>
      <t rank="1.0" pos="V">wapimaw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">see him.</trunc>
      <t rank="1.0" pos="V">wapimew</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">see a reflection of himself. Either physically or mentally, a characteristic. E.g. father to son.</trunc>
      <t rank="1.0" pos="V">wapimisow</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">see himself I the mirror.</trunc>
      <t rank="1.0" pos="V">wapimow</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD CW">have his/her shadow seen</trunc>
      <t rank="1.0" pos="V">cikâstêsin</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">disapprove of what s/he sees</trunc>
      <t rank="1.0" pos="V">macâpisin</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">the new moon is out or can be seen. hang new; be a new moon</trunc>
      <t rank="1.0" pos="V">oskakocin</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see strange sights</trunc>
      <t rank="1.0" pos="V">pîtotâpisin</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">fly away [Usually, a bird] and you are watching from behind. see it flying away from you. fly away [in the direction of the addressee]; fly onward</trunc>
      <t rank="1.0" pos="V">atimihâw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">run away in the distance [and you can see the person]. run away; run in the addressee's direction; run on ahead away from speaker</trunc>
      <t rank="1.0" pos="V">atimipahtâw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see such a distance</trunc>
      <t rank="1.0" pos="V">iskwâpiw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">be seen distinctly, clearly.</trunc>
      <trunc sources="CW">be clearly visible, look sharp and clear, be seen distinctly</trunc>
      <t rank="1.0" pos="V">kîhkânâkosiw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see clearly, see well; have good vision</trunc>
      <t rank="1.0" pos="V">nahâpiw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">go and watches; observe things, watch things; go to see people, try to see people</trunc>
      <t rank="1.0" pos="V">nitawâpahkêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">can be seen.</trunc>
      <trunc sources="CW">appear, come into view, be visible, be seen; be born</trunc>
      <t rank="1.0" pos="V">nôkosiw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">be alive to see one another</trunc>
      <t rank="1.0" pos="V">ocâpahtowak</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">be seen coming [by horse or vehicle].</trunc>
      <trunc sources="CW">come driving along; move here, drive here; be seen coming [by horse or vehicle]</trunc>
      <t rank="1.0" pos="V">pâpayiw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">be unable to see. be blind.</trunc>
      <trunc sources="CW">have a ruptured eye; have only one eye, be one-eyed; be unable to see, be blind; open his/her own eyes</trunc>
      <t rank="1.0" pos="V">pâskâpiw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD CW">be seen approaching</trunc>
      <t rank="1.0" pos="V">pêcinâkosiw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">can be seen with only the had sticking out. stick his/her own head out</trunc>
      <t rank="1.0" pos="V">sâkiskwêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">can be seen through. Animate. be transparent; be seen through [like a shawl can be seen through]</trunc>
      <t rank="1.0" pos="V">sâponâkosiw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see wrongly, see something else</trunc>
      <t rank="1.0" pos="V">wanâpamow</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see one another</trunc>
      <t rank="1.0" pos="V">wâpahtowak</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see him/herself</trunc>
      <t rank="1.0" pos="V">wâpamisow</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see one another</trunc>
      <t rank="1.0" pos="V">wâpamitowak</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see his/her own child; give birth to a child</trunc>
      <t rank="1.0" pos="V">wâpamâwasow</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see, have vision; [generally in negative phrases:] be blind</trunc>
      <t rank="1.0" pos="V">wâpiw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">be seen distinctly, clearly.</trunc>
      <trunc sources="CW">be clearly visible, look sharp and clear, be seen distinctly</trunc>
      <t rank="1.0" pos="V">kîhkânâkwan</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">can be seen. be seen.</trunc>
      <trunc sources="CW">appear, become visible, be seen</trunc>
      <t rank="1.0" pos="V">nôkwan</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD CW">be seen approaching</trunc>
      <t rank="1.0" pos="V">pêcinâkwan</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">there is brightness from the moon [can see everything]. N- moonlight.</trunc>
      <trunc sources="CW">be dawn, be first light; be moonlight so bright as to be light as day</trunc>
      <t rank="1.0" pos="V">kîsikâyâstêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD CW">project out of the ground</trunc>
      <t rank="1.0" pos="V">sâkaskitêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">be transparent. Inanimate. can see through it.</trunc>
      <trunc sources="CW">be transparent</trunc>
      <t rank="1.0" pos="V">sâpwâstêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.o. going away</trunc>
      <t rank="1.0" pos="V">atimâpamêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">track him.</trunc>
      <trunc sources="CW">come upon s.o.'s track, see s.o.'s footprints; track s.o.</trunc>
      <t rank="1.0" pos="V">mâtâhêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">see him readily.</trunc>
      <trunc sources="CW">see s.o. clearly, readily; see s.o. favourably</trunc>
      <t rank="1.0" pos="V">nahâpamêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">go and sees s.o., try to see s.o., go to visit s.o.</trunc>
      <t rank="1.0" pos="V">nitawâpamêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">go to see s.o., fetch s.o.</trunc>
      <t rank="1.0" pos="V">nâtawâpamêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">be alive to see s.o.</trunc>
      <t rank="1.0" pos="V">ocâpamêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.o. coming in the distance</trunc>
      <t rank="1.0" pos="V">otisâpamêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.o. pass</trunc>
      <t rank="1.0" pos="V">pimâpamêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">be seen coming to him. come to s.o.</trunc>
      <t rank="1.0" pos="V">pê-otihtêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">come and sees s.o.</trunc>
      <t rank="1.0" pos="V">pê-wâpamêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.o. at close range, nearby</trunc>
      <t rank="1.0" pos="V">pêswâpamêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.o. coming</trunc>
      <t rank="1.0" pos="V">pêtisâpamêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.o. coming</trunc>
      <t rank="1.0" pos="V">pêtâpamêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see through s.o.; take s.o.'s x-ray</trunc>
      <t rank="1.0" pos="V">sâpwâpamêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">see it. Animate. saw s.o. with a rip saw</trunc>
      <t rank="1.0" pos="V">tâskipohêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.o., witness s.o.</trunc>
      <t rank="1.0" pos="V">wâpamêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">make s.o. see</trunc>
      <t rank="1.0" pos="V">wâpihêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.o. as light in colour</trunc>
      <t rank="1.0" pos="V">wâpênêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.o. thus</trunc>
      <t rank="1.0" pos="V">isinawêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">go to see about s.o. [e.g. a sick person]; check on s.o.</trunc>
      <t rank="1.0" pos="V">nitawâpênawêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see [it/him] on or for s.o.</trunc>
      <t rank="1.0" pos="V">wâpahtamawêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">cry after him [as he is gone].</trunc>
      <trunc sources="CW">beg s.o. with tears; cry for s.o., mourn for s.o.; be sad to see s.o. go</trunc>
      <t rank="1.0" pos="V">mawîhkâtêw</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.t. thus; think s/he sees s.t.</trunc>
      <t rank="1.0" pos="V">isinam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">that is the way he saw it. look at s.t. thus</trunc>
      <t rank="1.0" pos="V">itâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">see and knows into the future.</trunc>
      <trunc sources="CW">use the shaking lodge; know future events; find out about s.t. through the spirits</trunc>
      <t rank="1.0" pos="V">kosâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD CW">push s.t. under [out of sight]</trunc>
      <t rank="1.0" pos="V">kotâwinam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see [s.t. as] daylight</trunc>
      <t rank="1.0" pos="V">kîsikânam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">see it readily. be quick to see it.</trunc>
      <trunc sources="CW">see s.t. clearly, readily; see s.t. favourably</trunc>
      <t rank="1.0" pos="V">nahâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">go to see s.t., try to see s.t.</trunc>
      <t rank="1.0" pos="V">nitawâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">go to see s.t., fetch s.t.</trunc>
      <t rank="1.0" pos="V">nâtawâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">be alive to see s.t.</trunc>
      <t rank="1.0" pos="V">ocâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.t. arriving, have lived long enough to see s.t., live long enough to see s.t. again</trunc>
      <t rank="1.0" pos="V">otisâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.t. clearly</trunc>
      <t rank="1.0" pos="V">pakaskinam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">be seen coming to it. come to s.t., arrive at his/her destination</trunc>
      <t rank="1.0" pos="V">pê-otihtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">come to see it. come and sees s.t.</trunc>
      <t rank="1.0" pos="V">pê-wâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.t. at close range, nearby</trunc>
      <t rank="1.0" pos="V">pêswâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.t. coming</trunc>
      <t rank="1.0" pos="V">pêtisâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.t. coming</trunc>
      <t rank="1.0" pos="V">pêtâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">see through it e.g.. A microscope. see through s.t., take a x-ray of s.t.</trunc>
      <t rank="1.0" pos="V">sâpwâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">be able to see [all the way to] s.t.</trunc>
      <t rank="1.0" pos="V">têpâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">see it.</trunc>
      <trunc sources="CW">see s.t., witness s.t.</trunc>
      <t rank="1.0" pos="V">wâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.t. as almost white</trunc>
      <t rank="1.0" pos="V">wâpênam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">see s.t. by the light it gives</trunc>
      <t rank="1.0" pos="V">wâsinam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="CW">look back and sees s.t.; look back at s.t.</trunc>
      <t rank="1.0" pos="V">âpasâpahtam</t>
   </tg>
   </mg>
   <mg>
   <tg xml:lang="crk">
      <trunc sources="MD">decorate it to be seen or noticed.</trunc>
      <trunc sources="CW">put a mark on s.t.</trunc>
      <t rank="1.0" pos="V">kiskinawâcihtâw</t>
   </tg>
   </mg>
</e>

```

**Comment by Madoshakalaka** [comment]
The arbitrariness of `engcrk.xml` begs the question of an alternative English keyword matching algorithm. I don't think I can **just** use all the Cree words inside `t` tag either (Imagine showing the user 50 results when he searches for "see")

**Comment by Madoshakalaka** [comment]
I think the students version searched for English keyword containment, ranks the result by definition shortness (which is a indication of relatedness), and shows the top ten results.

I remember it worked quite good. How about using that instead?

**Comment by Madoshakalaka** [comment]
The students version wasn't using `engcrk.xml` though. I employed `engcrk.xml` with the wish it gives more precise English matching. But turns out it won't give a big help.

**Comment by aarppe** [comment]
Since we do not have a proper English-to-Cree version of the dictionary, we have instead uses the English words in the glosses (with nouns and verbs lemmatized) as kaywords, under which the Cree lexical entries are listed. That is how this is done in the paper version of Wolvengrey's dictionary as well. It does provide long lists, but that is just how things are.

**Comment by aarppe** [comment]
In the old itwêwina, when you search with _see_ you get.a long list of all Cree lexical entries that have _see_ in their English translatoin (organized under POS). We don't really have any good way of pruning that, and in some sense it is actually good to see that _seeing_ in Cree is expressed by so many words.

**Comment by Madoshakalaka** [comment]
@aarppe Sure! I'll sort results by keyword matching fraction and have a limit of 10 or so. 

"s/he **see**" -> 3/8
"be able to **see** [all the way to] s.t." -> 3/36

May need your feedback later on the perceptual accurateness.

**Comment by aarppe** [comment]
We have had a long discussion about how to sort such searches. To do that properly, one would need manual scrutiny of the English-to-Cree direction, and that isn't happening anytime soon.

One option that has been considered is ranking the Cree lexical entries based on the number of words (or the character length) of their English translations.

Another option that we could follow is using corpus-based frequency of the Cree lexical entries, at least to organize them into brackets.

A third option, which could be complement the previous, would be to use an intersection of multiple search words (but maybe that's for later).

**Comment by aarppe** [comment]
In practice, I don't think we ought to limit what you get. But you would want to rule out determiners and articles (_a_, _an_, _the_). Even for frequent prepositions and adverbs, it is useful to see how they are used (maybe excluding _of_ that's really only grammatically used)

**Comment by Madoshakalaka** [comment]
Get it!

**Comment by aarppe** [comment]
And in creating `engcrk.xml`, ruling out determiners has already been done, so you can use the English search key words in their entirety (this should be done afterwards by the programmers, but earlier by the linguists).

**Comment by Madoshakalaka** [comment]
update: this is now fixed in my development branch. though the results is not sorted (sorting has a separate issue). Will push to Sapir soon

**Comment by eddieantonio** [comment]
Fixed as of: #180
